patches/minecraft/net/minecraft/network/NetHandlerPlayServer.java.patch

```diff
@@ -1,6 +1,22 @@
 --- ../src-base/minecraft/net/minecraft/network/NetHandlerPlayServer.java
 +++ ../src-work/minecraft/net/minecraft/network/NetHandlerPlayServer.java
-@@ -611,7 +611,10 @@
+@@ -571,9 +571,12 @@
+ 
+                 if (!this.field_147369_b.func_175149_v())
+                 {
+-                    ItemStack itemstack1 = this.field_147369_b.func_184586_b(EnumHand.OFF_HAND);
+-                    this.field_147369_b.func_184611_a(EnumHand.OFF_HAND, this.field_147369_b.func_184586_b(EnumHand.MAIN_HAND));
+-                    this.field_147369_b.func_184611_a(EnumHand.MAIN_HAND, itemstack1);
++                	List<ItemStack> l = net.minecraftforge.common.ForgeHooks.onSwapHand(this.field_147369_b);
++                	if(!l.isEmpty())
++                	{
++                		this.field_147369_b.func_184611_a(EnumHand.OFF_HAND, l.get(0));
++                    	this.field_147369_b.func_184611_a(EnumHand.MAIN_HAND, l.get(1));
++                	}
```

Comment:
Not what I meant.
Leave the lines that actually swap the stacks as is (no indentation, nothing). Then add `if FH.onSwapHand(...) {` before it and `}` after it. No need for any `List` return or anything.